catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/federation/impl/SortedFederationStrategy.java

```diff
@@ -332,4 +337,34 @@ public void run() {
       offsetResultQueue.closeResultQueue();
     }
   }
+
+  /**
+   * Logs unhandled Throwable, adds processing details, and closes the result queue when Errors
+   * (e.g. NoClassDefFoundError) and RuntimeExceptions are thrown from the wrapped runnable.
+   */
+  static class QueryResponseRunnableMonitor implements Runnable {
+
+    final Runnable wrapped;
+
+    final QueryResponseImpl queryResponse;
+
+    QueryResponseRunnableMonitor(Runnable runnable, QueryResponseImpl queryResponse) {
+      wrapped = runnable;
+      this.queryResponse = queryResponse;
+    }
+
+    @Override
+    public void run() {
+      try {
+        wrapped.run();
+      } catch (Throwable t) {
+        LOGGER.debug("Unhandled exception while watching query response runnable.", t);
+        queryResponse
+            .getProcessingDetails()
+            .add(new ProcessingDetailsImpl("unknown", new RuntimeException(t)));
```

Comment:
✏️ Might be good to create a custom subclass of `RuntimeException` for use here, especially as it would allow for the code that catches the exception to handle it differently from other RTExceptions.

Reply:
I think I would need to be exporting that exception for it to help other modules.  If I export from the standard framework, then that would create coupling to that module.  Though I think throwing a `FederationException` would make sense here since it is from the federation strategy which throws `FederationException`s.